performance/eclipselink.perf.test/src/test/java/org/eclipse/persistence/testing/perf/jpa/tests/basic/JPAReadAbstract.java

```diff
@@ -72,25 +92,22 @@ public void testReadEntity() {
         }
     }
 
-    private void prepareData(EntityManager em) {
-        try {
-            em.getTransaction().begin();
-            for (int i = 1; i <= getMasterSize(); i++) {
-                MasterEntity masterEntity = new MasterEntity(i, "Master name " + i);
-                em.persist(masterEntity);
-                for (int j = 1; j <= getDetailSize(); j++) {
-                    DetailEntity detailEntity = new DetailEntity(i * DETAIL_ID_STEP + j, "Detail name " + j, masterEntity);
-                    masterEntity.getDetails().add(detailEntity);
-                    em.persist(detailEntity);
-                }
-            }
-            em.getTransaction().commit();
-        } catch (Throwable e) {
-            throw new RuntimeException(e);
-        } finally {
-            if (em.getTransaction().isActive()) {
-                em.getTransaction().rollback();
+    private synchronized void prepareData(EntityManager em) {
+        AbstractSession session = em.unwrap(AbstractSession.class);
+        EmulatedConnection conn = (EmulatedConnection)session.getAccessor().getConnection();
+        Vector<DatabaseField> masterFields = new Vector<>(session.getDescriptor(MasterEntity.class).getAllFields());
```

Comment:
No need to use Vector here

Reply:
Due a `org.eclipse.persistence.internal.sessions.ArrayRecord#ArrayRecord(java.util.Vector, org.eclipse.persistence.internal.helper.DatabaseField[], java.lang.Object[])` is `java.util.Vector` needed.